content/en/agent/faq/linux-agent-2022-key-rotation.md

```diff
@@ -0,0 +1,136 @@
+
+<div class="alert alert-info">
+<strong>Note</strong>: This DOES NOT affect the functionality of already running Agents, and only limits the ability to install or upgrade to a newer version of the Agent. Also, this doesn't affect Dockerized Linux Agents, Windows, or macOS Agents.
+</div>
+
+## Install methods that automatically trust the new GPG key
+
+Your host automatically trusts the new key (no further action is required) if you're using one of the following install methods:
+
+- [Agent install script][5] v1.6.0+ (released Jul 26, 2021)
+- [Chef cookbook][6] v4.11.0+ (released Aug 10, 2021)
+- [Ansible role][7] v4.10.0+ (released May 25, 2021)
+- [Puppet module][8] v3.13.0+ (released Aug 11, 2021)
+- [SaltStack formula][9] v3.4+ (released Aug 12, 2021)
+- [Heroku buildpack][10] v1.26+ (released May 26, 2021)
+- [Elastic Beanstalk][11] config templates updated as of Mar 29, 2021 or later (should contain `DATADOG_RPM_KEY_FD4BF915.public` under `gpgkey`)
+- Containerized Agents (Docker/Kubernetes) for any version
+- Windows/MacOS Agents for any version
+
+Additionally, if you are using the DEB package and Agent v7.31.0+, your hosts should have the `datadog-signing-keys` package installed, which automatically adds the new key (no further action is needed).
+
+If you're unsure if a host trusts the new signing key, you can [check](#check-if-a-host-trusts-the-new-gpg-key).
+
+For hosts running older versions of the install methods listed above or older versions of the DEB package, Datadog recommends updating the install method to the latest version. Alternatively Debian and Ubuntu users can update the Agent to version 7.31.0+. Otherwise, the key can be [manually updated](#manual-update).
+
+## What happens if I don't trust the new key before it is rotated?
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Trying to install or upgrade Agent packages without trusting the new key results in `NO_PUBKEY` errors. This applies to both newly released and existing versions of the Agent.
+
+```
+The following signatures couldn't be verified because the public key is not available: NO_PUBKEY
+```
+
+{{% /tab %}}
+{{% tab "RedHat/CentOS/SUSE" %}}
+
+Installing new versions of the Agent released since April 2022 causes `NOKEY` errors. Older versions of the Agent can still be installed.
+
+```
+The GPG keys listed for the "Datadog, Inc." repository are already installed but they are not correct for this package.
+Check that the correct key URLs are configured for this repository.
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Manual update
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Run the following commands on the host:
+
+```bash
+$ curl -o /tmp/DATADOG_APT_KEY_F14F620E https://keys.datadoghq.com/DATADOG_APT_KEY_F14F620E.public
+$ sudo apt-key add /tmp/DATADOG_APT_KEY_F14F620E
+$ cat /tmp/DATADOG_APT_KEY_F14F620E | sudo gpg --import --batch --no-default-keyring --keyring /usr/share/keyrings/datadog-archive-keyring.gpg
```

Comment:
This is the scenario which assumes that the keyring file doesn't exist yet. I just tested it and running the command fails when the keyring doesn't exist. We should add `touch /usr/share/keyrings/datadog-archive-keyring.gpg` before this command.